Mathlib/Algebra/Module/LinearMap/Polynomial.lean

```diff
@@ -0,0 +1,336 @@
+/-
+Copyright (c) 2024 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+import Mathlib.Data.MvPolynomial.Monad
+import Mathlib.LinearAlgebra.Charpoly.ToMatrix
+import Mathlib.LinearAlgebra.Dimension.Finrank
+import Mathlib.LinearAlgebra.Matrix.Charpoly.Univ
+
+/-!
+# Characteristic polynomials of linear families of endomorphisms
+
+The coefficients of the characteristic polynomials of a a linear family of endomorphisms
+are homogeneous polynomials in the parameters.
+This result is used in Lie theory
+to establish the existence of regular elements and Cartan subalgebras,
+and ultimately a well-defined notion of rank for Lie algebras.
+
+In this file we prove this result about characteristic polynomials.
+Let `L` and `M` be modules over a nontrivial commutative ring `R`,
+and let `φ : L →ₗ[R] Module.End R M` be a linear map.
+Let `b` be a basis of `L`, indexed by `ι`.
+Then we define a multivariate polynomial with variables indexed by `ι`
+that evaluates on elements `x` of `L` to the characteristic polynomial of `φ x`.
+
+## Main declarations
+
+* `Matrix.toMvPolynomial M i`: the family of multivariate polynomials that evaluates on `c : n → R`
+  to the dot product of the `i`-th row of `M` with `c`.
+* `LinearMap.toMvPolynomial b₁ b₂ f`: a version of `Matrix.toMvPolynomial` for linear maps `f`
+  with respect to bases `b₁` and `b₂` of the domain and codomain.
+`Matrix.toMvPolynomial M i` is the sum of the monomials `C (M i j) * X j`.
```

Comment:
This line looks like is has gone astray (I guess either drop it or move it up two lines and ident).